pkg/suggest/rules_yaml_test.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestGetStructKeys(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    interface{}
+		expected map[string][]string
+	}{
+		{
+			name:     "int",
+			input:    1,
+			expected: map[string][]string{},
+		},
+		{
+			name:     "string",
+			input:    "field1",
+			expected: map[string][]string{},
+		},
+		{
+			name:     "bool",
+			input:    true,
+			expected: map[string][]string{},
+		},
+		{
+			name:     "map",
+			input:    map[string]string{},
+			expected: map[string][]string{},
+		},
+		{
+			name:     "pointer",
+			input:    &struct{}{},
+			expected: map[string][]string{},
+		},
+		{
+			name: "map with struct values no yaml tags",
+			input: map[string]struct {
+				field1 string
+				field2 int
+			}{
+				"key1": {"value1", 1},
+			},
+			expected: map[string][]string{},
+		},
+		{
+			name: "map with struct values with yaml tags",
+			input: map[string]struct {
+				field1 string `yaml:"field1"`
+				field2 int    `yaml:"field2"`
+			}{
+				"key1": {"value1", 1},
+			},
+			expected: map[string][]string{"_": []string{"field1", "field2"}},
+		},
+		{
+			name:     "string",
+			input:    "not a struct",
+			expected: map[string][]string{},
+		},
+		{
+			name:     "struct with no fields",
+			input:    struct{}{},
+			expected: map[string][]string{},
+		},
+		{
+			name: "anonymous struct with fields but no yaml tags",
+			input: struct {
+				field1 string
+				field2 string
+			}{},
+			expected: map[string][]string{},
+		},
+		{
+			name: "anonymous struct with fields",
+			input: struct {
+				field1 string `yaml:"field1"`
+				field2 string
+			}{},
+			expected: map[string][]string{"_": []string{"field1"}},
+		},
+		{
+			name: "anonymous struct with nested struct",
+			input: struct {
+				field1 string `yaml:"field1"`
+				nested struct {
+					field2 string `yaml:"field2"`
+				}
+			}{},
+			expected: map[string][]string{"_": []string{"field1", "field2"}},
+		},
+		{
+			name: "anonymous struct with nested struct with no yaml tags",
+			input: struct {
+				field1 string `yaml:"field1"`
+				nested struct {
+					field2 string
+				}
+			}{},
+			expected: map[string][]string{"_": []string{"field1"}},
+		},
+		{
+			name: "anonymous struct with nested struct with pointer",
+			input: struct {
+				field1 string `yaml:"field1"`
+
+				nested *struct {
+					field2 string `yaml:"field2"`
+				}
+			}{},
+			expected: map[string][]string{"_": []string{"field1", "field2"}},
+		},
+		{
+			name: "anonymous struct with nested struct with pointer with no yaml tags",
+			input: struct {
+				field1 string `yaml:"field1"`
+				nested *struct {
+					field2 string
+				}
+			}{},
+			expected: map[string][]string{"_": []string{"field1"}},
+		},
+		{
+			name:  "okteto manifest",
+			input: model.Manifest{},
+			expected: map[string][]string{
+				"forward.Forward":            []string{"localPort", "remotePort", "name", "labels"},
```

Comment:
nit: according to simplifycompositelit(default) redundant type from array, slice, or map composite literal

here and all the []string inside maps

```suggestion
				"forward.Forward":            {"localPort", "remotePort", "name", "labels"},
```

Reply:
Updated, thanks ✅